src/javascript/esutil.ts

```diff
@@ -199,6 +199,109 @@ export function getPropertyValue(
   }
 }
 
+export function getDescription(jsdocAnn: jsdoc.Annotation): string|undefined {
+  if (jsdocAnn.description) {
+    return jsdocAnn.description;
+  }
+  // These tags can be used to describe a field.
+  // e.g.:
+  //    /** @type {string} the name of the animal */
+  //    this.name = name || 'Rex';
+  const tagSet = new Set(['public', 'private', 'protected', 'type']);
+  for (const tag of jsdocAnn.tags) {
+    if (tagSet.has(tag.title) && tag.description) {
+      return tag.description;
+    }
+  }
+}
+
+export function propertiesFromGettersAndSetters(
+    body: Array<babel.Method|babel.Property>, document: JavaScriptDocument): ScannedProperty[] {
+  const properties: ScannedProperty[] = [];
+  const accessors = new Map<
+      string,
+      {getter?: babel.Method, setter?: babel.Method}>();
+
+  for (const member of body) {
+    if (!babel.isMethod(member)) {
+      continue;
+    }
+
+    // TODO: we dont support statics yet
+    if (babel.isClassMethod(member) && member.static) {
+      continue;
+    }
+
+    if (!babel.isIdentifier(member.key) || !member.key.name) {
+      continue;
+    }
+
+    const name = member.key.name;
+
+    if (member.kind !== 'get' && member.kind !== 'set') {
+      continue;
+    }
+
+    let accessor = accessors.get(name);
+
+    if (!accessor) {
+      accessor = {};
+      accessors.set(name, accessor);
+    }
+
+    if (member.kind === 'get') {
+      accessor.getter = member;
+    } else {
+      accessor.setter = member;
+    }
+  }
+
+  for (const [name, accessor] of accessors) {
+    let comment: string | undefined = undefined;
+    const member = (accessor.getter || accessor.setter)!;
+
+    if (accessor.getter) {
+      comment = getAttachedComment(accessor.getter);
+    }
+
+    if (accessor.setter && !comment) {
```

Comment:
here we kind of want to say, "use what you can from the getter, otherwise get it from the setter". we're not quite doing that here, though. do you think its worth basically `||`ing all the annotation-based stuff? (e.g. `getReturnFromAnnotation(getter) || getReturnFromAnnotation(setter)`)

Reply:
Doing that kind of merging is hard. What if they have incompatible information? I think only taking the getter's info if there's comments on both is a simple policy that's easy to understand, so this is fine to me.